core-web/apps/dotcms-ui/src/app/portlets/dot-edit-page/seo/components/dot-edit-page-state-controller-seo/dot-edit-page-state-controller-seo.component.ts

```diff
@@ -226,13 +291,38 @@ export class DotEditPageStateControllerSeoComponent implements OnChanges {
             live: !pageState.page.liveInode
         };
 
+        const enumMode = DotPageMode[mode.toLocaleUpperCase()] as DotPageMode;
+
         return {
             label: this.dotMessageService.get(`editpage.toolbar.${mode}.page`),
-            value: DotPageMode[mode.toLocaleUpperCase()],
+            value: {
+                id: enumMode,
+                showDropdownButton: this.shouldShowDropdownButton(enumMode, pageState)
+            },
             disabled: disabled[mode]
         };
     }
 
+    /**
+     * Check if the dropdown button should be shown
+     *
+     * @private
+     * @param {string} mode
+     * @param {DotPageRenderState} pageState
+     * @return {*}  {boolean}
+     * @memberof DotEditPageStateControllerSeoComponent
+     */
+    private shouldShowDropdownButton(mode: DotPageMode, pageState: DotPageRenderState): boolean {
+        const shouldModeShowDropdown: Record<DotPageMode, boolean> = {
+            [DotPageMode.EDIT]:
+                this.featureFlagEditURLContentMapIsOn && Boolean(pageState.params.urlContentMap),
+            [DotPageMode.PREVIEW]: true, // No logic involved, always show,
+            [DotPageMode.LIVE]: false // Don't show for live
+        };
```

Comment:
Can we define this object in the `ngOnInit` and delete the function?

Reply:
I don't think we can, because pageState can change, but I got rid of the object.